homeassistant/components/tellduslive/__init__.py

```diff
@@ -78,6 +78,19 @@
     hass.data[INTERVAL_TRACKER] = async_track_time_interval(
         hass, client.update, interval)
 
+    dev_reg = await hass.helpers.device_registry.async_get_registry()
+    for hub in client.get_hubs():
```

Comment:
This is doing I/O in the event loop. That's not allowed.

I also realize there's more places this is happening. We should fix this before next release. The tellduslive library isn't async safe at all.

Reply:
Methods on the session/client instance should be scheduled on the executor thread pool.

Reply:
This is what I mean with `@Throttle`

https://github.com/home-assistant/home-assistant/blob/a439690bd793392fa79151601425ce322fb4497d/homeassistant/components/sensor/trafikverket_weatherstation.py#L107

Reply:
What I mean is that I don't know why @molobrakos implemented the `track_point_in_utc_time` instead of letting hass do the polling with a reasonable `@Throttle`.

Reply:
The tellduslive entities don't use the home assistant builtin polling. Instead the polling is done from the tellduslive component. That looks good, since we don't want unnecessary calls to the api. I don't see a reason to change that.

Reply:
We're still in async context. We have to schedule this on the executor thread pool. Use `hass.async_add_executor_job`.

Reply:
It's on this line we need the executor.

Reply:
Or even better on 142.